Cargo.toml

```diff
@@ -13,6 +13,7 @@ readme = "README.md"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 [dependencies]
+serde = { version = "1.0.144", features = ["derive"], optional = true }
```

Comment:
In general when adding `serde` we try to avoid including the `derive` feature as it's  lead to slower compile time. See this [this section about derive in clap documentation](https://docs.rs/clap/latest/clap/_faq/index.html#when-should-i-use-the-builder-vs-derive-apis) about this.

Without the `derive` feature, we have to `impl Serialize` by hand. I think we can just copy and paste the code generated by the `#[derive(Serialize)`] macro.  To see the generated code we can use [cargo expand](https://github.com/dtolnay/cargo-expand).

There are many example of crate that add a serde feature without the derive, for instance  [Uuid](https://github.com/uuid-rs/uuid/blob/main/src/external/serde_support.rs)  or the [time](https://github.com/time-rs/time/tree/main/src/serde) crate.

Reply:
Thanks for the suggestions! I agree with you, let me apply changes and get back to you.